docs/changelog/4_x.rst

```diff
@@ -0,0 +1,467 @@
+So if there are multiple properties that should be searchable, they can all be
+specified in that one index:
+
+.. code-block:: sdl
+
+    type Item {
+      required available: bool {
+        default := false;
+      };
+      required name: str;
+      required description: str;
+
+      index fts::index on ((
+        fts::with_options(
+          .name,
+          language := fts::Language.eng
+        ),
+        fts::with_options(
+          .description,
+          language := fts::Language.eng
+        )
+      ));
+    }
+
+The above schema declares both ``name`` and ``description`` as searchable
+fields:
+
+.. code-block:: edgeql-repl
+
+  edgedb> with res := (
+  .......   select fts::search(Item, 'trick or treat', language := 'eng')
+  ....... )
+  ....... select res.object {name, description, score := res.score}
+  ....... order by res.score desc;
+  {
+    default::Item {
+      name: 'Candy corn',
+      description: 'It\'s a great Halloween treat',
+      score: 0.30396354,
+    },
+  }
+
+
+Multiranges
+-----------
+
+We've made it easier to work with ranges by adding ``multirange`` as one of
+the datatypes. Multiranges consist of one or more ranges and allow expressing
+intervals that are not contiguous. Multiranges are automatically normalized to
+contain non-overlapping ranges that are ordered according to their boundaries.
+All the usual range functions like ``overlaps`` or ``contains`` work with any
+combination of ranges and multiranges providing more flexibility in
+expressions.
+
+
+Extensions
+==========
+
+auth
+----
+
+The new ``auth`` extension adds a full authentication service that runs
+alongside your database instance saving you the hassle of having to learn and
+implement the intricacies of OAuth or secure password storage.
+
+- OAuth Integration: Seamlessly authenticate with GitHub, Google, Apple, and
+  Azure/Microsoft.
+- Email & Password Support: Includes robust email+password authentication with
+  reset password functionality.
+- Easy Configuration: Set up via our configuration system.
+- Hosted UI: Use our hosted authentication UI to quickly add authentication to
+  your app.
+
+When a user signs up, we create a new object of type ``ext::auth::Identity``,
+which you can link to in your own schema. We then provide you with a token that
+can be set as the global ``ext::auth::client_token`` which will automatically
+populate another computed global called ``ext::auth::ClientTokenIdentity``
+which you can use directly in your access policies, or in your own globals.
+
+.. code-block:: sdl
+
+    using extension auth;
+
+    module default {
+        global current_customer := (
+            assert_single((
+                select Customer { * }
+                filter .identity = global ext::auth::ClientTokenIdentity
+            ))
+        );
+
+        type Customer {
+            required text: str;
+            required identity: ext::auth::Identity;
```

Comment:
Can we put an exclusive constraint on this instead and drop the assert_single in the global?